code/game/machinery/kitchen/cooking_machines/_appliance.dm

```diff
@@ -0,0 +1,652 @@
+	if (cooking_objs.len)
+		var/string = "Contains..."
+		for (var/a in cooking_objs)
+			var/datum/cooking_item/CI = a
+			string += "-\a [CI.container.label(null, CI.combine_target)], [report_progress(CI)]</br>"
+		usr << string
+	else
+		usr << span("notice","It is empty.")
+
+/obj/machinery/appliance/proc/report_progress(var/datum/cooking_item/CI)
+	if (!CI || !CI.max_cookwork)
+		return null
+
+	if (!CI.cookwork)
+		return "It is cold."
+	var/progress = CI.cookwork / CI.max_cookwork
+
+	if (progress < 0.25)
+		return "It's barely started cooking."
+	if (progress < 0.75)
+		return span("notice","It's cooking away nicely.")
+	if (progress < 1)
+		return span("notice", "<b>It's almost ready!</b>")
+
+	var/half_overcook = (CI.overcook_mult - 1)*0.5
+	if (progress < 1+half_overcook)
+		return span("soghun","<b>It is done !</b>")
+	if (progress < CI.overcook_mult)
+		return span("warning","It looks overcooked, get it out!")
+	else
+		return span("danger","It is burning!!")
+
+/obj/machinery/appliance/update_icon()
+	if (!stat && cooking_objs.len)
+		icon_state = on_icon
+
+	else
+		icon_state = off_icon
+
+/obj/machinery/appliance/verb/toggle_power()
+	set src in view()
+	set name = "Toggle Power"
+	set category = "Object"
+
+	if (!isliving(usr))
+		usr << "Ghosts aren't allowed to toggle power switches"
+		return
+
+	if (!Adjacent(usr))
+		if (!istype(usr, /mob/living/silicon))
+			usr << "You can't reach the power switch from there, get closer!"
+			return
+
+	if (stat & POWEROFF)//Its turned off
+		stat &= ~POWEROFF
+		if (usr)
+			usr.visible_message("[usr] turns the [src] on", "You turn on the [src]")
+	else //Its on, turn it off
+		stat |= POWEROFF
+		if (usr)
+			usr.visible_message("[usr] turns the [src] off", "You turn off the [src]")
+	playsound(src, 'sound/machines/click.ogg', 40, 1)
+	update_icon()
+
+/obj/machinery/appliance/proc/choose_output()
+	set src in view()
+	set name = "Choose output"
+	set category = "Object"
+
+	if(output_options.len)
+
+		var/choice = input("What specific food do you wish to make with \the [src]?") as null|anything in output_options+"Default"
+		if(!choice)
+			return
+		if(choice == "Default")
+			selected_option = null
+			usr << "<span class='notice'>You decide not to make anything specific with \the [src].</span>"
+		else
+			selected_option = choice
+			usr << "<span class='notice'>You prepare \the [src] to make \a [selected_option] with the next thing you put in. Try putting several ingredients in a container!</span>"
+
+//Handles all validity checking and error messages for inserting things
+/obj/machinery/appliance/proc/can_insert(var/obj/item/I, var/mob/user)
+	if(!dropsafety(I))
+		return 0
+
+	// We are trying to cook a grabbed mob.
+	var/obj/item/weapon/grab/G = I
+	if(istype(G))
+
+		if(!can_cook_mobs)
+			user << "<span class='warning'>That's not going to fit.</span>"
+			return 0
+
+		if(!isliving(G.affecting))
+			user << "<span class='warning'>You can't cook that.</span>"
```

Comment:
Is it intentional that you can cook only living mobs here ?
Wouldnt it make sense to be able to cook both ?
Why would the machine care if some mob is dead or alive if it can fit inside ?

Also if you keep it, it would be nice to have a machine specific message for that, because you you dont "cook" with a mixer.